### DISCUSSION TITLE
Can I only use one worker thread to process requests?

### DISCUSSION THREAD

**Discussion by bobfang1992** [fastapi/fastapi#7139] [Questions]
Hi, for my use case I am wrapping up a Windows COM object to a HTTP service. The difficulty I am facing is that typically the COM API object can only be used in one thread and there are very complicated rules on how to use them in a multi-threading environment. It seems to me for every requests coming in fastapi typically choose to create new threads to process these requests (I am only using  the synced requests processor in my application --> think flask). I am just wondering for fastapi is there a way for me to limit this behavior and ask the app to create only one worker thread and stick to it?

Thansk!

**Comment by tiangolo** [comment]
I imagine you solved your problem, thanks for coming back to close the issue! ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
